packages/components/src/utils/floating-components.ts

```diff
@@ -0,0 +1,371 @@
+const isInView = (el: Element) => {
+	const { top, bottom, left, right } = el.getBoundingClientRect();
+	const { innerHeight, innerWidth } = window;
+
+	let outTop = top < 0;
+	let outBottom = bottom > innerHeight;
+	let outLeft = left < 0;
+	let outRight = right > innerWidth;
+
+	// We need to check if it was already outside
+	const outsideY = el.hasAttribute('data-outside-vy');
+	const outsideX = el.hasAttribute('data-outside-vx');
+	const parentRect = el?.parentElement?.getBoundingClientRect();
+
+	if (parentRect) {
+		if (outsideY) {
+			const position = el.getAttribute('data-outside-vy');
+			if (position === 'top') {
+				outTop = parentRect.top - (bottom - parentRect.bottom) < 0;
+			} else {
+				outBottom =
+					parentRect.bottom + (parentRect.top - top) > innerHeight;
+			}
+		}
+
+		if (outsideX) {
+			const position = el.getAttribute('data-outside-vx');
+			if (position === 'left') {
+				outLeft = parentRect.left - (right - parentRect.right) < 0;
+			} else {
+				outRight =
+					parentRect.right + (parentRect.left - left) > innerWidth;
+			}
+		}
+	}
+
+	return {
+		outTop,
+		outBottom,
+		outLeft,
+		outRight
+	};
+};
+
+export interface DBDataOutsidePair {
+	vx?: 'left' | 'right';
+	vy?: 'top' | 'bottom';
+}
+export const handleDataOutside = (el: Element): DBDataOutsidePair => {
+	const { outTop, outBottom, outLeft, outRight } = isInView(el);
+	let dataOutsidePair: DBDataOutsidePair = {};
+
+	if (outTop || outBottom) {
+		dataOutsidePair = { vy: outTop ? 'top' : 'bottom' };
+		el.setAttribute('data-outside-vy', dataOutsidePair.vy!);
```

Comment:
```suggestion
		el.dataset.outsideVy = dataOutsidePair.vy!;
```

Why don't you use the `dataset` API for setting `data-` attributes?

Reply:
It didn't work for angular. But I solved it with: `el.dataset["outsideVy"]`

Reply:
fixed